### ISSUE TITLE
yarn installation of self hosted git+ssh repos fail when there's a message returned from gitlab

### ISSUE THREAD

**Issue by mmucklo** [yarnpkg/yarn#3325]
yarn 0.23.4
macOs 10.12.4

When working with a "self-hosted" gitlab repo (such as talked about in #513), there's a case where gitlab is setup to output a warning message before returning the contents.

In certain settings gitlab server could possibly return a header prior to every interaction like:

```
*******************************************
Matt's Gitlab server, proceed at your own risk
*******************************************
abcde1234 refs/heads/blah
bdbadf4321 refs/heads/blah2
```

In these cases the following error is seen:

     TypeError: Cannot read property 'split' of undefined

When using a git+ssh://git@something/... etc. url

Stack trace:

```
at Function.parseRefs (...node/v6.10.3/lib/node_modules/yarn/lib/util/git.js:519:20)
at ...node/v6.10.3/lib/node_modules/yarn/lib/util/git.js:450:24
at next (native)
at step (node/v6.10.3/lib/node_modules/yarn/node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30
at node/v6.10.3/lib/node_modules/yarn/node_modules/babel-runtime/helpers/asyncToGenerator.js:28:13
at process._tickCallback (internal/process/next_tick.js:109:7)
```

**Comment by rally25rs** [comment]
I'm curious if this works in NPM.

We could possibly just ignore lines in the response that don't "look like refs" with some kind of regex, but that seems like a can of worms. The custom message could still "look like a ref" and pass a regex check.

**Comment by rally25rs** [comment]
@mmucklo could you take a look at this PR ☝️ and see if it fits your needs?

**Comment by bestander** [comment]
A PR with a test is welcome.

**Comment by rally25rs** [comment]
@bestander I did submit a PR with a test ;) #3351 
I had hoped @mmucklo would take a look at it, but :tumbleweeds:

**Comment by mmucklo** [comment]
@rally25rs - sorry I've been out of town.  I should be able to test the patch on Monday if not later today...

**Comment by mmucklo** [comment]
Fixed as of yarn v0.25.2
